README.md

```diff
@@ -22,6 +22,297 @@ development:
+the database.
+
+Here's an example using a Geos default factory:
+
+```ruby
+RGeo::ActiveRecord::SpatialFactoryStore.instance.tap do |config|
+  # By default, use the GEOS implementation for spatial columns.
+  config.default = RGeo::Geos.factory_generator
+
+  # But use a geographic implementation for point columns.
+  config.register(RGeo::Geographic.spherical_factory(srid: 4326), geo_type: "point")
+end
+```
+
+The default spatial factory for geographic columns is `RGeo::Geographic.spherical_factory`.
+The default spatial factory for cartesian columns is `RGeo::Cartesian.preferred_factory`.
+You do not need to configure the `SpatialFactoryStore` if these defaults are ok.
+
+More information about configuration options for the `SpatialFactoryStore` can be found in the [rgeo-activerecord](https://github.com/rgeo/rgeo-activerecord#spatial-factories-for-columns) docs.
+
+### Reading and Writing Spatial Columns
+
+When you access a spatial attribute on your ActiveRecord model, it is given to
+you as an RGeo geometry object (or nil, for attributes that allow null
+values). You can then call the RGeo api on the object. For example, consider
+the MySpatialTable class we worked with above:
+
+```rb
+record = MySpatialTable.find(1)
+point = record.lonlat                  # Returns an RGeo::Feature::Point
+p point.x                              # displays the x coordinate
+p point.geometry_type.type_name        # displays "Point"
+```
+
+The RGeo factory for the value is determined by how you configured the
+ActiveRecord class, as described above. In this case, we explicitly set a
+spherical factory for the `:lonlat` column:
+
+```rb
+factory = point.factory                # returns a spherical factory
+```
+
+You can set a spatial attribute by providing an RGeo geometry object, or by
+providing the WKT string representation of the geometry. If a string is
+provided, the adapter will attempt to parse it as WKT and
+set the value accordingly.
+
+```rb
+record.lonlat = 'POINT(-122 47)'  # sets the value to the given point
+```
+
+If the WKT parsing fails, the value currently will be silently set to nil. In
+the future, however, this will raise an exception.
+
+```rb
+record.lonlat = 'POINT(x)'         # sets the value to nil
+```
+
+If you set the value to an RGeo object, the factory needs to match the factory
+for the attribute. If the factories do not match, the adapter
+will attempt to cast the value to the correct factory.
+
+```rb
+p2 = factory.point(-122, 47)       # p2 is a point in a spherical factory
+record.lonlat = p2                 # sets the value to the given point
+record.shape1 = p2                 # shape1 uses a flat geos factory, so it
+                                   # will cast p2 into that coordinate system
+                                   # before setting the value
+record.save
+```
+
+If, however, you attempt to set the value to the wrong type-- for example,
```

Comment:
nit: spacing after `type`. also maybe `If you attempt to set the value to the wrong type, such as setting a linestring attribute to a point value, you will...` for better flow